src/serializers/type_serializers/dict.rs

```diff
@@ -92,8 +117,17 @@ impl TypeSerializer for DictSerializer {
                             SerMode::Json => self.key_serializer.json_key(&key, extra)?.into_py_any(py)?,
                             _ => self.key_serializer.to_python(&key, None, None, extra)?,
                         };
-                        let value =
-                            value_serializer.to_python(&value, next_include.as_ref(), next_exclude.as_ref(), extra)?;
+                        let value = if extra.sort_keys {
+                            let sorted_value = sort_dict_recursive(py, &value)?;
```

Comment:
Could we not do a recursive sort here, but sort `new_dict` specifically after we've built it here?